### PR TITLE
Feature/issue 4698 rxjava2 support

### PR DESCRIPTION
**Pull request by ber4444**
### PR checklist

- [X] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [X] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Implements https://github.com/swagger-api/swagger-codegen/issues/4698# including fixes for the "2 minor issues" found by @wing328.

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java [hunk 1]
```diff
@@ -84,9 +89,17 @@ public String getHelp() {
...
+            LOGGER.warn("You specified both RxJava versions 1 and 2 but they are mutually exclusive. Defaulting to v2.");
+        } else if (additionalProperties.containsKey(USE_RX_JAVA)) {
             this.setUseRxJava(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA).toString()));
         }
+        if (additionalProperties.containsKey(USE_RX_JAVA2)) {
+            this.setUseRxJava2(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA2).toString()));
+        }
+	if (!useRxJava && !useRxJava2) {
+	    additionalProperties.put(DO_NOT_USE_RX, true);
+	}
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: The above 3 lines is missing a space in the indentation.

Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: Please use 4-space instead of tab.

Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: Also shall we replace these 3 lines with the following 1-liner?
```
	    additionalProperties.put(DO_NOT_USE_RX, doNotUseRx);
```
to ensure additional property has the latest `doNotUseRx` value (true or false)

Comment (ber4444) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: That would not work. You can try it with a config such as 
```
{
  "library": "retrofit2",
  "artifactId": "swagger-locator2-retrofit2-rx",
  "useRxJava2": false,
  "hideGenerationTimestamp": true
}
```
and check to see that the results are incorrect with your proposed one liner. My version is correct, however.

Also, feel free to change tab to space but there are other tabs in the file, e.g. right at line 19, which were done by others.

Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: I'll perform a bit more tests and let you know how it goes.

Comment (ber4444) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java]: Hi, what's the status of this PR?

**Comment by ber4444** [comment]
@wing328 what's the status of this PR?

**Comment by wing328** [comment]
@ber4444 how urgent do you want this change to be merged into master? We're preparing the 2.2.2 release and want to freeze changes.

This is not to say your PR still has issues but just that we want to avoid last-minute change (if possible) before the release.

**Comment by ber4444** [comment]
I would not say it's urgent but it is certainly good if we can release it.

**Comment by wing328** [comment]
OK. We'll merge it after the 2.2.2 release.

**Comment by wing328** [comment]
@ber4444 thanks for the PR, which has been merged into master.

**Pull request merged**